### PR TITLE
Expose the backoff limit

### PR DESCRIPTION
**Pull request by fao89**
Depends-On: https://github.com/openstack-k8s-operators/openstack-ansibleee-operator/pull/356

### CODE REVIEW

File: pkg/deployment/deployment.go [hunk 1]
```diff
@@ -199,12 +199,21 @@ func (d *Deployer) ConditionalDeploy(
...
 		if ansibleEE.Status.JobStatus == ansibleeev1.JobStatusFailed {
+			severity := condition.SeverityWarning
+			errorReason := condition.ErrorReason
+			errorMsg := fmt.Sprintf("attempt #%d - execution.name %s execution.namespace %s execution.status.jobstatus: %s", ansibleEE.Status.failed, ansibleEE.Name, ansibleEE.Namespace, ansibleEE.Status.JobStatus)
+			ansibleCondition := ansibleEE.Status.Conditions.Get(condition.JobReadyCondition)
+			if ansibleCondition.Reason == condition.JobReasonBackoffLimitExceeded {
+				severity = condition.SeverityError
+				errorReason = condition.JobReasonBackoffLimitExceeded
+				errorMsg = fmt.Sprintf("backoff limit reached for execution.name %s execution.namespace %s execution.status.jobstatus: %s", ansibleEE.Name, ansibleEE.Namespace, ansibleEE.Status.JobStatus)
+			}
```
Comment (fao89) [inline pkg/deployment/deployment.go:210]: expose attempts in the status message

**Comment by fao89** [comment]
recheck

**Comment by fao89** [comment]
recheck

**Review by slagle — COMMENTED** [review]
+2

**Review by rebtoor — APPROVED** [review]

**Pull request merged**